core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -1131,6 +1132,36 @@ private RelNode mockCountOver(String table,
     assertThat(s, isLinux(expected));
   }
 
+  @Test void testDeserializeInvalidOperatorName() {
+    final FrameworkConfig config = RelBuilderTest.config().build();
+    final RelBuilder builder = RelBuilder.create(config);
+    final RelNode rel = builder
+        .scan("EMP")
+        .project(
+            builder.field("JOB"),
+            builder.field("SAL"))
+        .aggregate(
+            builder.groupKey("JOB"),
+            builder.max("max_sal", builder.field("SAL")),
+            builder.min("min_sal", builder.field("SAL")))
+        .project(
+            builder.field("max_sal"),
+            builder.field("min_sal"))
+        .build();
+    final RelJsonWriter jsonWriter = new RelJsonWriter();
+    rel.explain(jsonWriter);
+    // mock a non exist SqlOperator
+    String relJson = jsonWriter.asString().replace("\"name\": \"MAX\"", "\"name\": \"MAXS\"");
+    try {
+      deserializeAndDumpToTextFormat(getSchema(rel), relJson);
+      fail();
+    } catch (Exception e) {
+      assertThat(e.getMessage(),
+          isLinux("org.apache.calcite.runtime.CalciteException: "
+              + "No operator for 'MAXS' with kind: 'MAX', syntax: 'FUNCTION'"));
+    }
```

Comment:
Maybe we can use JUnit's `assertThrows` utility here?
Some examples can be found in org.apache.calcite.util.mapping.MappingTest.

Reply:
Sure